gateway-contracts/tasks/deployment/contracts.ts

```
@@ -62,6 +63,75 @@ async function deployContractImplementation(
   return proxyAddress;
 }
 
+async function deployMultisigSmartAccount(
+  name: string,
+  { run, ethers }: HardhatRuntimeEnvironment,
+  owners: string[],
+  threshold: number,
+) {
+  // Get a deployer wallet
+  const deployerPrivateKey = getRequiredEnvVar("DEPLOYER_PRIVATE_KEY");
+  const deployer = new Wallet(deployerPrivateKey).connect(ethers.provider);
+
+  // Deploy a new Safe contract
+  const safeFactory = await ethers.getContractFactory("Safe", deployer);
+  const safe = await safeFactory.deploy();
+  const safeAddress = await safe.getAddress();
+
+  // Deploy a new SafeProxyFactory contract
+  const safeProxyFactoryFactory = await ethers.getContractFactory("SafeProxyFactory", deployer);
+  const safeProxyFactory = await safeProxyFactoryFactory.deploy();
+
+  // Prepare the setup transaction data
+  if (owners.length === 0) {
```

we should avoid having default logic like this in order to prevent unwanted behaviors in case of misuse

so : 
- if the list it empty, the task should fail (you can throw an error here if you want but the contract should also revert)
- if someone wants to use the deployer as the owner, than whoever runs the task should provide the deployer in the list
- if having the deployer as the single owner really makes the task more convenient to use (for ZWS, for test purposes, ..) , then better to add an optional boolean in the tasks with default to false that will do this

---

Yes, I agree, thanks! 👍